ui/app/AppLayouts/Chat/views/private/MembersSelectorBase.qml

```diff
@@ -54,41 +47,32 @@ InlineSelectorPanel {
             return true
         }
 
-        function isPastedProfileLinkToContact(pubkey) {
-            return root.pastedChatKey === pubkey
-        }
-
         filters: [
-            ExpressionFilter {
+            FastExpressionFilter {
                 enabled: root.edit.text !== "" && root.pastedChatKey == ""
                 expression: {
                     root.edit.text // ensure expression is reevaluated when edit.text changes
                     return _suggestionsModel.searchPredicate(model.displayName, model.localNickname, model.alias)
                 }
+                expectedRoles: ["displayName", "localNickname", "alias"]
             },
-            ExpressionFilter {
+            FastExpressionFilter {
                 expression: {
                     root.model.count // ensure expression is reevaluated when members model changes
                     return _suggestionsModel.notAMemberPredicate(model.pubKey)
                 }
+                expectedRoles: ["pubKey"]
             },
-            ExpressionFilter {
-                enabled: root.pastedChatKey != ""
-                expression: {
-                    root.pastedChatKey // ensure expression is reevaluated when members model changes
-                    return _suggestionsModel.isPastedProfileLinkToContact(model.pubKey)
-                }
+            ValueFilter {
+                roleName: "pubKey"
+                value: root.pastedChatKey
+                enabled: root.pastedChatKey !== ""
             }
         ]
 
-        proxyRoles: ExpressionRole {
-            name: "title"
-            expression: model.localNickname || model.displayName || model.alias
-        }
-
         sorters: StringSorter {
-            roleName: "title"
-            numericMode: true
+            roleName: "preferredDisplayName"
```

Comment:
Using the `preferredDisplayName` model role instead of an extra `ExpressionRole`